tests/handlers/aws/test_outputs.py

```diff
@@ -0,0 +1,190 @@
+        {
+            "messageType": "DATA_MESSAGE",
+            "owner": "000000000000",
+            "logGroup": group_name,
+            "logStream": stream_name,
+            "subscriptionFilters": ["a-subscription-filter"],
+            "logEvents": collected_log_events,
+        }
+    )
+
+    data_gzip = gzip.compress(data_json.encode("UTF-8"))
+    data_base64encoded = base64.b64encode(data_gzip)
+
+    return {"awslogs": {"data": data_base64encoded}}, collected_log_event_ids
+
+
+@pytest.mark.integration
+class TestLambdaHandlerLogstashOutputSuccess(TestCase):
+    def setUp(self) -> None:
+        LocalStackContainer.IMAGE = "localstack/localstack:1.1.0"
+        lst = LocalStackContainer()
+        lst.with_env("EAGER_SERVICE_LOADING", 1)
+        lst.with_services("s3", "logs", "sqs")
+        self.localstack = lst.start()
+
+        session = boto3.Session(region_name="eu-west-1")
+        self.s3_client = session.client("s3", endpoint_url=self.localstack.get_url())
+        self.logs_client = session.client("logs", endpoint_url=self.localstack.get_url())
+        self.sqs_client = session.client("sqs", endpoint_url=self.localstack.get_url())
+
+        self.logstash_http_port = 5043
+        lgc = LogstashContainer(port=self.logstash_http_port)
+        lgc.with_env("CONFIG_STRING", '''\
+            input {{
+              http {{
+                port => {logstash_http_port}
+                codec => json_lines
+              }}
+            }}
+
+            output {{ stdout {{}} }}
+            '''.format(logstash_http_port=self.logstash_http_port))
+        self.logstash = lgc.start()
+
+        self.fixtures = {
+                "cw_log_1": load_file_fixture('cloudwatch-log-1.json'),
+                "cw_log_2": load_file_fixture('cloudwatch-log-2.json')
+        }
+
+        group_name = f"{type(self).__name__}-source-group"
+        stream_name = f"{type(self).__name__}-source-stream"
+
+        _create_cloudwatch_logs_group(self.logs_client, group_name=group_name)
+        g = _create_cloudwatch_logs_stream(self.logs_client,
+            group_name=group_name, stream_name=stream_name)
+        cloudwatch_group_arn = g['arn']
+
+        _event_to_cloudwatch_logs(
+            self.logs_client,
+            group_name=group_name,
+            stream_name=stream_name,
+            messages_body=[self.fixtures["cw_log_1"] + self.fixtures["cw_log_2"]],
+        )
+
+        self.group_name = group_name
+        self.stream_name = stream_name
+
+        config_content = load_file_fixture('config.yaml')
+        self.config = Template(config_content).substitute(dict(
+            CloudwatchLogStreamARN=cloudwatch_group_arn,
+            LogstashURL=self.logstash.get_url()
+        ))
+        print(self.config)
+
+        _upload_content_to_bucket(
+            client=self.s3_client,
+            content=self.config,
+            content_type="text/plain",
+            bucket_name=f"{type(self).__name__}-config-bucket",
```

Comment:
👍

Reply:
One note, I'd consider adding a `lower()` call somewhere since bucket names are strictly lowercase (ref: https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html). Things work now but I wouldn't be surprised if newer versions of localstack would enforce this.

Reply:
Created a dedicated helper and used `lower` in this case, thanks for catching it!